lib/samples/create_mobile_geodatabase/create_mobile_geodatabase_sample.dart

```diff
@@ -0,0 +1,353 @@
+                      // until state is ready.
+                      Visibility(
+                        visible: !(_ready),
+                        child: SizedBox.expand(
+                          child: Container(
+                            color: Colors.white30,
+                            child: const Center(
+                              child: CircularProgressIndicator(),
+                            ),
+                          ),
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+                // Display the number of features added and a button to view the table.
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(10, 0, 10, 0),
+                  child: Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      Text(
+                        'Number of features added: $_featureCount',
+                      ),
+                      OutlinedButton(
+                        style: OutlinedButton.styleFrom(
+                          shape: const RoundedRectangleBorder(
+                            borderRadius: BorderRadius.zero,
+                          ),
+                          side: const BorderSide(
+                            color: Colors.grey,
+                          ),
+                        ),
+                        onPressed: _displayTable,
+                        child: const Text(
+                          'View table',
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+                // Display a button to create and share the mobile geodatabase.
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(0, 10, 0, 10),
+                  child: ElevatedButton.icon(
+                    onPressed: _shareGeodatabaseUri,
+                    icon: const Icon(
+                      Icons.edit,
+                    ),
+                    label: const Text(
+                      'Create and share mobile geodatabase',
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    //_mapViewController.dispose();
+    _geodatabase?.close();
+    _geodatabase = null;
+    _featureTable = null;
+    super.dispose();
+  }
+
+  // When the map view is ready, create a map and set the viewpoint.
+  void onMapViewReady() {
+    _map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISTopographic);
+    _mapViewController.arcGISMap = _map;
+    _mapViewController.setViewpoint(
+      Viewpoint.withLatLongScale(
+        latitude: 41.5,
+        longitude: -100.0,
+        scale: 100000000.0,
+      ),
+    );
+    // Create the mobile geodatabase with a feature table to track
+    // location history.
+    _createGeodatabase();
```

Comment:
This should be `await _createGeodatabase();` and the `_createGeodatabase()` method should return a `Future<void>`. That will make sure the geodatabase is ready before it is used.